en/services/trajectory.md

```diff
@@ -0,0 +1,66 @@
+# Path Planning (Trajectory) Protocol
+
+The path planning protocol (a.k.a. trajectory interface) is a general-purpose protocol for enabling dynamic path planning from an external system (e.g. a companion computer).
+
+The protocol is primarily intended for cases where constraints on the path to a destination are unknown or may change dynamically, but it can also be used for any other path management activities.
+Examples include: *obstacle avoidance* when following a preplanned mission, determining paths for self forming/healing swarms, offloading geofence management to a companion computer, etc.
+
+
+## Overview
+
+The system that requires path-planning sends a message defining its current position and desired trajectory.
+The path planning software analyses the desired route and, if required (by the use case), sends back a stream of messages containing setpoints for a new path.
+
+{% mermaid %}
+sequenceDiagram;
+    participant Autopilot
+    participant Companion
+    Autopilot->>Companion: Desired path/trajectory
+    Companion->>Companion: Calculate best trajectory
+    Companion-->>Autopilot: Target setpoint (if required)
+{% endmermaid %}
+
+Vehicles are expected to move towards the most current setpoint from the remote system.
+If no new setpoints arrive before a vehicle reaches the last received setpoint, it may proceed along whatever path is currently demanded by the flight stack (e.g. a mission, return to land, etc.)
+
+The protocol defines two messages, either of which may be used to represent the desired trajectory and/or target setpoint:
+* [TRAJECTORY_REPRESENTATION_WAYPOINTS](../messages/common.md#TRAJECTORY_REPRESENTATION_WAYPOINTS) - Trajectory represented as an array of up-to 5 *waypoints* in the *local frame*.
+* [TRAJECTORY_REPRESENTATION_BEZIER](../messages/common.md#TRAJECTORY_REPRESENTATION_BEZIER) - Trajectory represented using an array of up-to 5 bezier points in the local frame.
```

Comment:
Thanks. This library is supposed to be purely about the protocol definition - ie it is abstract rather than talking about what a particular autopilot might do (although I have an implementation section at the bottom that says PX4 only uses the waypoints representation).

As designer of this interface, have you any insight to when a system might prefer to use a Bezier representation of the desired trajectory?

Reply:
if someone wants to send any spline representation to the autopilot, `TRAJECTORY_REPRESEMTATION_BEZIER` has the appropriate fields such as the time horizon